src/App.tsx

```diff
@@ -1,61 +1,128 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { ChangeEvent, FormEvent, useState } from 'react';
+import { getUserById } from './utils/getUserById';
+import { getMaxId } from './utils/getMaxId';
+
+const todosWithUser: Todo[] = todosFromServer.map(todo => {
+  return {
+    ...todo,
+    user: getUserById(todo.userId),
+  };
+});
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosWithUser);
+
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [selectedUser, setSelectedUser] = useState(0);
+  const [hasUserError, setHasUserError] = useState(false);
+
+  const handleSetTitle = (event: ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value.trimStart());
+
+    if (hasTitleError) {
+      setHasTitleError(false);
+    }
+  };
+
+  const handleSetUser = (event: ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUser(Number(event.target.value));
+
+    if (hasUserError) {
+      setHasUserError(false);
+    }
+  };
+
+  const handleSetTodos = () => {
+    const newTodo = {
+      id: getMaxId(todos),
+      title: title.trim(),
+      completed: false,
+      userId: selectedUser,
+      user: getUserById(selectedUser),
```

Comment:
The `getUserById` function is used to get the user object by `selectedUser`. Ensure that this function handles cases where a user might not be found, as it could lead to undefined behavior if the user ID does not exist in the users list.